File: basics/prime.cpp
#include<iostream>
using namespace std;

int main(){
    int n;
    bool flag=false;
    cout<<"enter a no.";
    cin>>n;
    for(int i=2; i*i<n; i++){
        if(n%i==0){
            flag=true;
            break;
        }
    }
        if(flag==true){
            cout<<n<<"No is not prime";
        }else{
            cout<<n<<"No is prime";
        }
    return 0;
}

File: basics/prime.java
import java.util.Scanner;
public class prime{
    public static void main(String args[]){
    boolean flag= false;
    System.out.println("enter a no.");
        Scanner s = new Scanner(System.in);
        int n= s.nextInt();
        for(int i=2;i<n;i++){
            if(n%i==0){
                flag= true;
                break;
            }
        }
        if(flag==true){
            System.out.println("this is no is not prime");
        }else{
            System.out.println("this no is prime");
        }
        s.close();
    }
}